ttg/ttg/parsec/ttg.h

```diff
@@ -1450,36 +1484,98 @@ namespace ttg_parsec {
     }
 
     template <ttg::ExecutionSpace Space>
-    static parsec_hook_return_t device_static_op(parsec_task_t* parsec_task) {
-      static_assert(derived_has_device_op());
-
-      int dev_index;
-      double ratio = 1.0;
+    static parsec_hook_return_t device_static_evaluate(parsec_task_t* parsec_task) {
 
       task_t *task = (task_t*)parsec_task;
-      parsec_execution_stream_s *es = task->tt->world.impl().execution_stream();
+      if (task->dev_ptr->gpu_task == nullptr) {
+
+        //std::cout << "device_static_op: task " << parsec_task << std::endl;
+
+        /* set up a device task */
+        parsec_gpu_task_t *gpu_task;
+        /* PaRSEC wants to free the gpu_task, because F***K ownerships */
+        gpu_task = static_cast<parsec_gpu_task_t*>(std::calloc(1, sizeof(*gpu_task)));
+        PARSEC_OBJ_CONSTRUCT(gpu_task, parsec_list_item_t);
+        gpu_task->ec = parsec_task;
+        gpu_task->task_type = 0; // user task
+        //gpu_task->load = 1;    // TODO: can we do better?
```

Comment:
the load will be filled up by the default-time-estimate function, which by default will use the compute capacity of the target device to compute in normalized units: 1 load on the fastest device, x load on slower devices where x is the performance ratio of that device w.r.t. the fastest device. In short you should not need to do this commented out thing at all. You can provide better time_estimate functions (see how it's done in ptg) if you want to be more precise w.r.t. the amount of flops individual tasks incur.

Reply:
I removed the comment.